bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/AbstractComponent.java

```diff
@@ -120,8 +123,11 @@ public void setConfigSeen() {
      */
     public CompletableFuture<@Nullable Void> start(MqttBrokerConnection connection, ScheduledExecutorService scheduler,
             int timeout) {
-        return channels.values().parallelStream().map(cChannel -> cChannel.start(connection, scheduler, timeout))
-                .collect(FutureCollector.allOf());
+        @NonNull
+        Collector<@NonNull CompletableFuture<@Nullable Void>, @NonNull Set<@NonNull CompletableFuture<@Nullable Void>>, @NonNull CompletableFuture<@Nullable Void>> allOfCollector = FutureCollector
+                .allOf();
+        return channels.values().stream().map(cChannel -> cChannel.start(connection, scheduler, timeout))
+                .collect(allOfCollector);
```

Comment:
`cChannel.start()` has some bits `synchronized` anyways, so this is good change.

Other changes are to remove null typing warnings